src/webpack/plugin/applyDefaults.js

```diff
@@ -0,0 +1,62 @@
+const startCase = require('lodash.startcase');
+const fs = require('fs');
+const path = require('path');
+
+const npmPackage = JSON.parse(fs.readFileSync(path.join(process.cwd(), 'package.json')));
+
+const applyDefaults = ({
+  primaryNavigationItems = [{
+    path: '/home',
+    label: 'Home',
+    contentExtension: 'home',
+    additionalContent: [
+      {
+        label: 'Home',
+        filePath: path.resolve(process.cwd(), 'README.md'),
+      },
+    ],
+  }, {
+    path: '/components',
+    label: 'Components',
+    contentExtension: 'doc',
+  }, {
+    path: '/tests',
+    label: 'Tests',
+    contentExtension: 'test',
+  }],
+  additionalSearchDirectories = [],
+  sideEffectImportFilePaths = [],
+  titleConfig = {
+    title: startCase(npmPackage.name.split('/').pop()),
+  },
+  defaultLocale,
+  defaultTheme,
+  defaultDirection,
+  faviconFilePath = path.join(__dirname, '..', 'terra-favicon', '32px', 'favicon.ico'),
+  extensionItems = [],
+  headHtml = [],
+  pathPrefix,
+  sourceFolder = 'src',
+  distributionFolder = 'lib',
+  enableDebugLogging = false,
+  additionalConfig = {},
+} = {}) => ({
```

Comment:
Probably not much different, but thoughts on using lodash.get's API for determine the fallback/default value?

Reply:
this method is nice in that it allows m to default specific entries if not defined and it doesn't introduce another dependency.